src/Controller/Api/ApiController.php

```diff
@@ -2,107 +2,81 @@
-            Response::HTTP_OK,
-            [
-                'Content-type' => 'application/json'
-            ]
-        );
+        return new JsonResponse($this->generateUrls($data, $urlGenerator));
     }
 
     /**
-     * Returns all resources of application.
+     * Returns information about client and writes it to log file.
      *
-     * @Route("/api", methods="GET", name="list_of_resources")
+     * @Route("/api/my-info", methods="GET", name="client_info")
      *
      * @param Request $request
      *
-     * @return Response
+     * @return JsonResponse
      */
-    public function actionIndex(Request $request): Response
+    public function actionClientInfo(Request $request): JsonResponse
     {
-        if (!$request->headers->get('Content-type: application/json')) {
-            throw new UnsupportedMediaTypeHttpException();
-        }
+        $clientInfo = [
+            'ip' => $request->getClientIps()[0],
+            'language' => $request->getPreferredLanguage(),
+            'browser' => $request->headers->get('User-Agent')
+        ];
+        $encoded = json_encode($clientInfo);
 
-        $data = $this->getDataFromFile();
-        $responseData = $this->prepareData($request, $data);
-
-        return new Response(
-            json_encode($responseData),
-            Response::HTTP_OK,
-            [
-                'Content-type' => 'application/json'
-            ]
-        );
+        file_put_contents('logs.txt', $encoded . PHP_EOL, FILE_APPEND | LOCK_EX);
+
+        return new JsonResponse($clientInfo);
     }
 
     /**
      * Returns all resources of application.
      *
      * @Route("/api/{product}", methods="GET", name="product_id")
      *
-     * @param Request $request
      * @param string $product
+     * @param UrlGeneratorInterface $urlGenerator
+     * @param TranslatorInterface $translator
      *
-     * @return Response
+     * @return JsonResponse
      */
-    public function actionProduct(Request $request, string $product): Response
-    {
-        if (!$request->headers->get('Content-type: application/json')) {
-            throw new UnsupportedMediaTypeHttpException();
-        }
-
+    public function actionProduct(
+        string $product,
+        UrlGeneratorInterface $urlGenerator,
+        TranslatorInterface $translator,
+    ): JsonResponse {
         $data = $this->getDataFromFile();
-        $uris = $this->prepareData($request, $data);
-        $neededProducts = array_filter($uris, function ($item) use ($product) {
-            return str_contains($item, $product);
-        });
+        $urls = $this->generateUrls($data, $urlGenerator);
+        $neededProducts = array_filter($urls, fn ($item) => str_contains($item, $product));
 
         if (empty($neededProducts)) {
             throw new NotFoundHttpException();
         }
 
         $neededProduct = $data[key($neededProducts)];
+        $translatedKeys = $this->translate($this->getTranslateIds(), $translator);
```

Comment:
Having two separate methods that must be running in a chain is not the best design.
It's much better to inline the code from getTranslateIds here, as it's easier to read the code without scrolling back and forth. Too small methods to follow SRP is a bad practice.

Reply:
I assumed that i recieve translate ids from database in getTranslateIds() method. So thats only the reason.

Reply:
But anyway thanks, i'm appreciate it